src/Microsoft.Health.Fhir.Shared.Web/Startup.cs

```diff
@@ -27,10 +27,12 @@ public Startup(IConfiguration configuration)
         public virtual void ConfigureServices(IServiceCollection services)
         {
             services.AddDevelopmentIdentityProvider(Configuration);
+            services.AddMemoryCache();
```

Comment:
>services.AddMemoryCache(); [](start = 12, length = 26)

I can see it's been used in tests, is it also requirement for non test cases?

Reply:
Yes. We inject the MemoryCache for template provider factory (which needs a cache to determine whether to load cache or pull remote templates) in the NuGet dependency

Reply:
As intermediary step, I think it would be nice to add comment on top of that code to state necessity of memory cache for TemplateProviderFactory.

But overall, I think TemplateProviderFactory need to provide extension for IServiceCollection to initialize all necessary dependencies. Same way as you have extension for ContainerRegistry in this PR. People who integrate this library shouldn't rely on secret knowledge (especially if there is chance it's gonna change in future)

Reply:
Agree. An registration extension is really helpful. Added:)